mitodl/micromasters
# Used virtualized scrolling for AutoComplete component

#### What are the relevant tickets?
Fixes #515 

#### What's this PR do?
Adds `Menu` and `List` from `material-ui` and modifies them and `AutoComplete` to increase performance and use virtualized scrolling.

#### Where should the reviewer start?

#### How should this be manually tested?
In general things should act exactly as they did before, except faster hopefully or at least not slower. Open up micromasters-ci in a separate tab and for each case compare functionality between it and your local instance which is on this branch.

First, go through all of the cases in the manual testing section of https://github.com/mitodl/micromasters/pull/382 and verify that it all works. Additionally check these cases (make sure to confirm the redux state for each selection):
 - You should be able to press down to select the first item in the menu. If you press enter you should select that item in the list. 
 - Type text to filter the results, then press down and select some other item in the list. If you press enter you should select that item in the list.
 - Type text to filter the results, press down, then click an item on the list. It should be selected successfully.
 - Type text then press enter. The first item in the results should be selected.

Other things to check for:
 - scrollbars should show up if and only if appropriate. You should be able to use them like any other list.
 - When you click on a autocomplete field with many items, like the country field, it should open about as quickly as any other autocomplete field on the page. It should not be slower than it was before.
 - You should be able to scroll to the bottom of the list and select the last item without any problem.
 - The popup height should be at most 300 pixels, and if less space is needed it should shrink to fit the items in the list.
 - The width should match the text field's width.

Bugs:
 - if you press the page down key a few times and then press down, it will continue scrolling. This is different from the previous behavior where the down key would move to the next selected item in the list, no matter how far down in the list you go.

#### What GIF best describes this PR or how it makes you feel?
http://i.giphy.com/zf3skxc4BFBqo.gif
